RestartManager.js

```diff
@@ -3,41 +3,54 @@ const NativeCodePush = require("react-native").NativeModules.CodePush;
 
 const RestartManager = (() => {
     let _allowed = true;
-    let _restartPending = false;
+    let _restartInProgress = false;
+    let _restartQueue = [];
 
     function allow() {
         log("Re-allowing restarts");
         _allowed = true;
-        
-        if (_restartPending) {
+
+        if (_restartQueue.length) {
             log("Executing pending restart");
-            restartApp(true);
+            restartApp(_restartQueue.shift(1));
         }
     }
-    
+
     function clearPendingRestart() {
-        _restartPending = false;
+        _restartQueue = [];
     }
-    
+
     function disallow() {
         log("Disallowing restarts");
         _allowed = false;
     }
 
-    function restartApp(onlyIfUpdateIsPending = false) {
-        if (_allowed) {
-            NativeCodePush.restartApp(onlyIfUpdateIsPending);
-            log("Restarting app");
-        } else {
+    async function restartApp(onlyIfUpdateIsPending = false) {
```

Comment:
If the customer wants to use this API to restart the app regardless of whether an update is pending or not, he would call this with `false`. If he only wants to restart if there is an actual update downloaded, he would call this with `true`.